### DISCUSSION TITLE
Editing TEDIT files

### DISCUSSION THREAD

**Discussion by blakemcbride** [Interlisp/medley#115] [Q&A] [answered]
Greetings,

Is there a command to edit a .TEDIT file directly without going through the directory browser?

Thanks!

Blake

**Comment by masinter** [comment, marked as answer]
https://github.com/Interlisp/history/tree/master/docs/Interlisp%20Manual/Lisp%20Library/Tedit 

(TEDIT "teditfile.tedit")
@Anzus has been working on documentation; not sure where in the medley repo the TEDIT documentation went.

**Reply by blakemcbride** [reply]
Thanks, however, if I:

1.  Create a text file in my home directory named test.tedit using VI and put "Hello World" in it.
2.  Then from within Medley, I do: (TEDIT "test.tedit")

Medley edits the string "test.tedit" rather than the contents of the file named "test.tedit".

unless I am misunderstanding something.

Thanks!

Blake

**Reply by blakemcbride** [reply]
I can do:  (TEDIT)
And then a GET from within TEDIT to get the file.  That works.  But can I just get the file as part of the (TEDIT) command?

Thanks!

Blake

**Reply by blakemcbride** [reply]
I see.  I need to type:  (TEDIT 'test.tedit)
it works perfectly.
The argument needs to be a symbol and not a string.
Thanks!
Blake

**Reply by masinter** [reply]
I was wondering what forces led John to raise the limit on the number of symbols from 2 to 3 to 4 byte's worth. It seemed like a stretch.
But Interlisp code often returns (interned) symbols (litatoms) and some bits of code expects them too.
Common Lisp introduced a PATHNAME datatype which had parsed components. 
I might even count it as a "bug" that TEDIT won't accept a string or a pathname for the name of the file.
